synapse/api/auth.py

```diff
@@ -629,7 +629,13 @@ def get_user_by_access_token(self, token, rights="access"):
         except AuthError:
             # TODO(daniel): Remove this fallback when all existing access tokens
             # have been re-issued as macaroons.
+            if self.hs.config.expire_access_token:
+                raise AuthError(
+                    401,
+                    "access token has expired"
+                )
```

Comment:
`raise` by itself will raise the previous exception raised, so:

```python
try:
    ...
except Exception:
    raise
```

is equivalent to:

```python
try:
    ...
except Exception as e:
    raise e
```

(which would be equivalent to not catching the exception).

Hence:

```python
except AuthError:
    if self.hs.config.expire_access_token:
        raise
    ret = yield self._look_up_user_by_access_token(token)
```

will effectively only catch the exception if `self.hs.config.expire_access_token` is `True`

(You need to be a little bit careful with using `raise` versus `raise e` if you call a function before re-raising the exception, as the function itself may raise and catch an exception.)